c7n/resources/sagemaker.py

```diff
@@ -101,6 +104,9 @@ def _augment(e):
             tags = client.list_tags(
                 ResourceArn=e['EndpointArn'])['Tags']
             e['Tags'] = tags
+
+            # normalize arn attribute for flat tagging
```

Comment:
rather than copying/duplicating the attribute could you add the arn's key as attribute to the resource_type and use that metadata when doing the generic sagemaker tag, there's also resource manager api support for getting a set of resource arns.